### PR TITLE
WRP-8078:Button: Add roundBorder prop making both sides of button fully round

### PR DESCRIPTION
**Pull request by hoeunryuLG**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
As a result of a discussion with the GUI team, it was decided to provide a boolean prop to have both the left and right sides of the button completely round. Therefore, a `roundBorder` boolean prop is added to achieve that.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Use
- `--button-height: $height`
- `border-radius: calc(--button-height / 2)`.
    
Use CSS custom property `--button-height` and LESS property as variable `$height`. CSS custom property `--button-height` is overridden wherever CSS property `height` is defined.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
add sampler/qa-sampler and screenshot tests for roundBorder prop

- sampler
  - add roundBorder prop in controls
- qa-sampler
  - add roundBorder variants in each `<Section>`s of `KitchenSink`
- screenshot tests
  - add roundBorder variants for the following
    - size
    - backgroundOpacity
    - icon
    - large text size

### Links
[//]: # (Related issues, references)
WRP-8078

### Comments
Enact-DCO-1.0-Signed-off-by: Hoeun Ryu <hoeun.ryu@lge.com>

### CODE REVIEW

File: samples/sampler/stories/qa/Button.js [hunk 1]
```diff
@@ -300,11 +329,23 @@ export const KitchenSink = () => (
...
+					Button
+				</Button>
 			</Section>
 
 			<Section title="Static Focus Effect Icon" size="50%">
 				<Button focusEffect="static" size="small" icon="play" alt="Small" />
+				<Button focusEffect="static" roundBorder size="small" icon="play" alt="Small Round Border" />
 				<Button focusEffect="static" size="large" icon="play" alt="Large" />
+				<Button focusEffect="static" roundBorder size="large" icon="play" alt="Large Round Border" />
+
```
Comment (juwonjeong) [inline samples/sampler/stories/qa/Button.js:348]: Please remove line
```suggestion
```

Comment (hoeunryuLG) [reply samples/sampler/stories/qa/Button.js:348]: I'll fix it

**Review by juwonjeong — APPROVED** [review]
LGTM

File: Button/Button.js [hunk 2]
```diff
@@ -166,6 +166,15 @@ const ButtonBase = kind({
 		 */
 		minWidth: PropTypes.bool,
 
+		/**
+		 * True if both sides of button are fully rounded.
+		 *
+		 * @type {Boolean}
+		 * @default false
```
[the remark this answers is not in the document]
Comment (hoeunryuLG) [reply Button/Button.js:173]: OK, I will fix it.

**Comment by hoeunryuLG** [comment]
A fix-up patch has been pushed to address the comments.

**Review by 0x64 — APPROVED** [review]

**Pull request merged**